templates/dart/lib/client.dart.twig

```diff
@@ -96,6 +96,10 @@ class Client {
           if(e.response == null) {
             throw {{spec.title | caseUcfirst}}Exception(e.message);
           }
+          if(responseType == ResponseType.bytes) {
+            final res = json.decode(utf8.decode(e.response.data));
```

Comment:
Why do we need to decode the image to UTF8 and not sure why json.decode is used

Reply:
I think I got it, this is to return the JSON error properly?

Reply:
Maybe we can have a nicer switch-case based on the *response* (not request header) content-type header?

Also note, your'e using `AppwriteException` instead of `{{spec.title | caseUcfirst}}Exception`

Reply:
1. AppwriteException hardcoded is Fixed
2. We don't need to check anything else for now, just when ResponseType is bytes, so I think its' fine? What do you suggest?

Reply:
I think we should change it to avoid future surprises. Check the response content-type header is the correct way to figure out which parsing to apply for the payload.

Reply:
got it